submit.go

```diff
@@ -41,69 +44,99 @@ var (
 	}
 )
 
+var (
+	configPath = "configuration.ini"
+)
+
 type (
-	Client struct {
-		OVH          *ovh.Client
-		lastPrintLog uint64
-		JobID        string
+	OVHConf struct {
+		Endpoint          string `ini:"endpoint"`
+		ApplicationKey    string `ini:"application_key"`
+		ApplicationSecret string `ini:"application_secret"`
+		ConsumerKey       string `ini:"consumer_key"`
 	}
 )
 
 // main ovh-spark-submit entry point
 func main() {
+	var err error
 
 	jobSubmitValue := ParsArgs()
 
-	var err error
-	ovh, err := ovh.NewDefaultClient()
+	conf, err := InitConf(configPath)
 	if err != nil {
-		log.Fatalf("Error while creating OVH Client: %s\nYou need to create an application; please visite this page https://api.ovh.com/createApp/ and create your ovh.conf file\n\t[default]\n\t; general configuration: default endpoint\n\tendpoint=ovh-eu\n\n\t[ovh-eu]\n\t; configuration specific to 'ovh-eu' endpoint\n\tapplication_key=my_app_key", err)
+		log.Fatalf("Unable to load conf: %s", err)
 	}
-	client := &Client{
-		OVH: ovh,
+
+	ovhConf := new(OVHConf)
+	if err := conf["ovh"].MapTo(ovhConf); err != nil {
+		log.Fatalf("Unable to load \"ovh\" conf: %s", err)
 	}
 
-	job := client.Submit(args.ProjectID, jobSubmitValue)
-	SetupCloseHandler(client)
-	//poll Status
-statusLoop:
-	for {
-		switch job.Status {
-		case JobStatusUNKNOWN, JobStatusSUBMITTED, JobStatusPENDING:
-			log.Printf("Job is %s", job.Status)
+	ovhClient, err := ovh.NewClient(
+		ovhConf.Endpoint,
+		ovhConf.ApplicationKey,
+		ovhConf.ApplicationSecret,
+		ovhConf.ConsumerKey,
+	)
+	if err != nil {
+		log.Fatalf("Error while creating OVH Client: %s", err)
+	}
 
-		case JobStatusCANCELLING, JobStatusTERMINATED, JobStatusFAILED, JobStatusCOMPLETED:
-			break statusLoop
+	if args.Upload != "" {
+		args.File = filepath.Clean(args.File)
+		splitFile := strings.Split(args.File, "/")
```

Comment:
"/" : what if you're not on a unix based system?

Reply:
if your are on windows it will not work

Reply:
Understood